### ISSUE TITLE
Mesa update broke Vulkan

### ISSUE THREAD

**Issue by urioTV** [NixOS/nixpkgs#320768]
### Describe the bug
Vulkan stopped working on my machine. OpenGL works OK. 

### Steps To Reproduce
1. Run app that is using Vulkan(example vkcube)
2. App does not run.

### Expected behavior
App should run without any issues

### Additional context
After update I couldn't be able to run games on my PC. I checked that whole Vulkan stopped working. 

```console
Selected GPU 0: AMD Radeon Graphics (RADV NAVI31), type: DiscreteGpu
vulkan: No DRI3 support detected - required for presentation
Note: you can probably enable DRI3 in your Xorg config
vulkan: No DRI3 support detected - required for presentation
Note: you can probably enable DRI3 in your Xorg config
vulkan: No DRI3 support detected - required for presentation
Note: you can probably enable DRI3 in your Xorg config
Could not find both graphics and present queues
```
That happens when I try to run vkcube. 

When I use mesa-git from Chaotic Nyx problem disappears so I'm pretty sure it's mesa's fault.

### Notify maintainers

@primeos @vcunat 

@K900 as you touched it last time

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.3-zen1, NixOS, 24.11 (Vicuna), 24.11.20240616.b60ebf5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/5w7nhyvrgsa81h0kyifmbl6j97hvlzh5-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
Are you on X11 or Wayland? Have you restarted your system after upgrading?

**Comment by urioTV** [comment]
```nix
hyprland = {
      url = "git+https://github.com/hyprwm/Hyprland?submodules=1&rev=fe7b748eb668136dd0558b7c8279bfcd7ab4d759";
    };
```

I'm using this as an input. That a 0.39.1 Hyprland version.

**Comment by K900** [comment]
Yeah, you need to set `inputs.nixpkgs.follows = "nixpkgs"` on that, or just use the nixpkgs package.

**Comment by urioTV** [comment]
Well It worked before without that but I'll try

**Comment by K900** [comment]
By accident, yes.

**Comment by K900** [comment]
Because mesa-git does weird things to libgbm. Anyway, resolved.
